### PR TITLE
Update FOSSA configuration for new requirements layout

### PR DESCRIPTION
**Pull request by robdiciuccio**
### SUMMARY
FOSSA is not accurately checking or failing on license issues with modules in the base requirements. This PR updates the FOSSA config to test `base` and `docker` requirements for license issues, and fails the build if any are found.

I've also made the FOSSA report public, so anyone should be able to view.

NOTE: there are currently dependencies in `base.txt` that do not meet Apache licensing requirements (e.g. `mysqlclient` and `psycopg2-binary`), so the check is expected to fail. 

### TEST PLAN
- FOSSA should run in CI and should fail on incompatible licenses

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Review by willbarrett — APPROVED** [review]
LGTM after rebase

**Comment by robdiciuccio** [comment]
Example public FOSSA report:
https://app.fossa.com/projects/custom+11342%2Fgithub.com%2Fapache%2Fincubator-superset/refs/branch/HEAD/c1b5d64859540a08ba74a41bfc3c9a54d506b2e4

File: .fossa.yml [hunk 1]
```diff
@@ -28,13 +28,15 @@ analyze:
...
+  - name: base
     type: pip
-    target: docs
-    path: docs
-  - name: .
+    target: .
+    path: .
+    options:
+      requirements: ./requirements/base.txt
+  - name: docker
```
Comment (craig-rueda) [inline .fossa.yml:37]: We don't need this check, as the Docker requirements are only used for the "dev" image (which is used in docker-compose)

**Review by craig-rueda — CHANGES_REQUESTED** [review]

File: .fossa.yml (hunk 1 above)
Comment (robdiciuccio) [reply .fossa.yml:37]: So, it looks like the "lean" Docker image is using `requirements/local.txt`? This is confusing.

Comment (craig-rueda) [reply .fossa.yml:37]: Yep, lean gets "local" and dev gets "docker". When I did this, "local" already existed, so I reused it. I added "docker.txt" in order to append a few more deps for docker-compose.

Comment (robdiciuccio) [reply .fossa.yml]: Ok, removed it for now. May need some reworking after https://github.com/apache/incubator-superset/pull/10875 settles.

**Review by craig-rueda — APPROVED** [review]

**Comment by robdiciuccio** [comment]
We should probably run the license check each time code is merged to master, as well as when the requirements files change.

**Pull request merged**